sympy/sets/tests/test_fancysets.py

```diff
@@ -295,6 +296,114 @@ def test_Range_set():
         assert r.inf == rev.inf and r.sup == rev.sup
         assert r.step == -rev.step
 
+    # test symbolic Range
+    n = symbols('n', integer=True)
+    r = Range(1, n)
+    assert r.subs(n, -2) == Range(0, 0, 1)
+    assert r.subs(n, 3) == Range(1, 3, 1)
+    ranges = [ Range(n, 10, 1), Range(n, -10, 1), Range(n, -10, -1),
```

Comment:
I would much rather see s systematic set of tests that show all possible meaningful values for the symbols. See the code that I sent you: when stepping by two you can either hit or miss the end point so test (1, 2, 2), (1, 3, 2), (1, 4, 2) and test reversal (2, 1, 2). Then negate: (-1, -2, -2), (-1, -3, -2), (-1, -4, -2), (-2, -1, -2). Then test that a==b gives empty range and c==0 raises error.

The logic of the tests is not transparent -- why did you select the number that you did?